modules/sdk-coin-sui/src/lib/transaction.ts

```diff
@@ -281,16 +281,20 @@ export class Transaction extends BaseTransaction {
   }
 
   static updateProperTransactionData(k: any): void {
-    // bcs deserializes number into Big Number, needs to convert them back to number
+    // bcs deserializes number into Big Number and removed '0x' prefix from addresses, needs to convert them back
     k.payTx.amounts = k.payTx.amounts.map((amount) => amount.toNumber());
     k.payTx.coins = k.payTx.coins.map((coin): SuiObjectRef => {
       return {
-        objectId: coin.objectId,
+        objectId: coin.objectId.startsWith('0x') ? coin.objectId : '0x'.concat(coin.objectId),
```

Comment:
thoughts on creating a helper function in `util` to contain this logic?

maybe something like:
```
/**
     Normalizes hex ids (addresses, object ids) to always contain the '0x' prefix.
**/
function normalizeHexId(id: string): string {
  ...
}

Reply:
updated